source/features/stop-redirecting-commit-pull-url-to-commit-url.tsx

```diff
@@ -0,0 +1,31 @@
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '../libs/features';
+
+function init(): void {
+	delegate(document, 'form#new_issue', 'submit', updateTextArea);
+	delegate(document, 'form.js-new-comment-form', 'submit', updateTextArea);
+}
+
```

Comment:
This does everything I suggested. You can move it to `libs/utils.ts`

```suggestion

const escapeRegex = (string: string) => string.replace(/[\\^$.*+?()[\]{}|]/g, '\\$&');
const prCommitRegex = new RegExp(`\\b${escapeRegex(location.origin)}[/][^/]+[/][^/]+[/]pull[/]\\d+[/]commits[/][0-9a-f]{7,40}\\b(?! \\]|\\))`, 'gi');

function preventPrCommitLinkBreak(comment: string) {
  return comment.replace(prCommitRegex, '[$& ]($&)')
}

```